source/common/common/key_value_store_base.cc

```diff
@@ -72,21 +71,23 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents) {
     if (checkForTtl(contents)) {
       uint64_t ttl_int;
       auto token = getToken(contents, error);
-      if (token.has_value() && absl::SimpleAtoi(token.value(), &ttl_int)) {
+      if (!token.has_value()) {
+        ENVOY_LOG(warn, error);
+        return false;
+      }
+      if (absl::SimpleAtoi(token.value(), &ttl_int)) {
         ttl.emplace(std::chrono::duration_cast<std::chrono::seconds>(
-            std::chrono::time_point<std::chrono::steady_clock>(std::chrono::seconds(ttl_int)) -
-            time_source_.monotonicTime()));
-      } else {
-        if (!token.has_value()) {
-          ENVOY_LOG(warn, error);
-        } else {
-          ENVOY_LOG(warn, "TTL was read from disk but failed to convert to integer");
+            std::chrono::time_point<std::chrono::system_clock>(std::chrono::seconds(ttl_int)) -
+            time_source_.systemTime()));
+        if (ttl <= std::chrono::seconds(0)) {
+          continue;
         }
+      } else {
+        ENVOY_LOG(warn, "TTL was read from disk but failed to convert to integer");
+        return false;
```

Comment:
I think you could do an early return to reduce some of this nesting, e.g. `if (!absl::SimpleAtoi(...)) { return false }`